### ISSUE TITLE
[no-explicit-any] Allow generic tuple type usage

### ISSUE THREAD

**Issue by be5invis** [typescript-eslint/typescript-eslint#1432]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-explicit-any": "error"
  }
}
```

```TS
function tuple<T extends any[]>(...args: T): T {
    return args;
}
```

**Expected Result**

Eslint accepts this code with no error.

**Actual Result**

Eslint rejects this code.

**Additional Info**

This sample is taken from the [official example](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-0.html#rest-elements-in-tuple-types) about tuple types. ESLint should accept this usage.

**Comment by armano2** [comment]
any is type unsafe, and this rule aims to prevent you from using this, in provided example you should use `unknown` instead.

> TypeScript 3.0 introduces a new top type unknown. unknown is the type-safe counterpart of any. Anything is assignable to unknown, but unknown isn’t assignable to anything but itself and any without a type assertion or a control flow based narrowing. Likewise, no operations are permitted on an unknown without first asserting or narrowing to a more specific type.

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-0.html#new-unknown-top-type

**Comment by be5invis** [comment]
`T extends unknown[]` seems working. Closed.
